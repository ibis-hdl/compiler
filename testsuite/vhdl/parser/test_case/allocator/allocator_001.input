-- subtype indication 
new BIT_VECTOR(1 to 3)

