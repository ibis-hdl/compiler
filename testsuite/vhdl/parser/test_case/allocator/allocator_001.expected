(allocator<v>
  (subtype_indication
    (type_mark<v>
      (name<v>
        (identifier
          BIT_VECTOR
        )
      )
    )
    (constraint<v>
      (index_constraint
        (discrete_range<v>
          (range<v>
            (simple_expression
              (term
                (primary<v>
                  (literal<v>
                    (numeric_literal<v>
                      (abstract_literal<v>
                        (decimal_literal
                          literal: 1, type: integer
                        )
                      )
                    )
                  )
                )
              )
            )
            (keyword
              TO
            )
            (simple_expression
              (term
                (primary<v>
                  (literal<v>
                    (numeric_literal<v>
                      (abstract_literal<v>
                        (decimal_literal
                          literal: 3, type: integer
                        )
                      )
                    )
                  )
                )
              )
            )
          )
        )
      )
    )
  )
)
